src/user.rs

```diff
@@ -852,6 +852,8 @@ impl UserNode {
         &mut self,
         mempool_peer: SocketAddr,
     ) -> Result<()> {
+        // TODO: having next_payment as part of the node is error-prone, it would be better to
```

Comment:
Yeah, this was a last minute change made to accommodate an exchange, and still uses this weird async construction and sending structure. Would you like to make this change as part of this MR @DaMatrix?

Reply:
Sure, will do